editoast/src/views/electrical_profiles.rs

```diff
@@ -48,12 +49,31 @@ async fn get(
     .unwrap()
 }
 
+#[post("/")]
+async fn post_electrical_profile(
+    db_pool: Data<DbPool>,
+    name: Query<String>,
+    data: Json<ElectricalProfileSetData>,
+) -> Result<Json<ElectricalProfileSet>> {
+    block(move || {
+        let mut conn = db_pool.get().expect("Failed to get DB connection");
+        let electrical_profile_set = ElectricalProfileSet::create_electrical_profile_set(
+            &mut conn,
+            name.into_inner(),
+            data.into_inner(),
+        )?;
+        Ok(Json(electrical_profile_set))
+    })
+    .await
+    .unwrap()
```

Comment:
`ElectricalProfileSet::create_electrical_profile_set` should take a `Data<DbPool>` argument and be responsible to block and get a PgConnection.

Reply:
Can I get the existing DbBool in `src/main.rs` ? Do I need to use `Pool::builder()` ?

![Screenshot_20230307_151159](https://user-images.githubusercontent.com/24652545/223447142-76254ed4-44f1-4a00-9b88-d840058d9e22.png)

Reply:
By the way, is it Okay to use the `view` ElectricalProfileSet here in `main.rs` ? I had to make `ElectricalProfileSet` public

Reply:
You can do this to get a `Data<Pool>`:

```rust
let pool = Data::new(Pool::builder().max_size(1).build(manager).unwrap());
```

-----------------

It's okay to use the view in the main but ideally, you would separate the view from the operations. For example, there are two module documents.

- The first one contains the operations you can perform.
- The second one contains the view that simply calls the operations.